### PR TITLE
[STORY] Add delete interview to the SDK

### PR DESCRIPTION
**Pull request by DfernandezNipo**
**Description**
GDPR: Mark an interview as deleted

**Implementation**
- Update SDK for GDPR Delete Interview feature. The SDK will use the Service to call the public API to allow deleting an Interview PI data.

### CODE REVIEW

File: Tests/Services/NfieldInterviewsServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+using Nfield.Models.NipoSoftware.Nfield.Manager.Api.Models;
+using Nfield.Services.Implementation;
+using Xunit;
+
+namespace Nfield.Services
+{
+    public class NfieldInterviewsServiceTests : NfieldServiceTestsBase
+    {
+        private readonly string _surveyId = Guid.NewGuid().ToString();
+        private readonly int _interviewId = new Random().Next(9999);
```
Comment (carloscoceraperez) [inline Tests/Services/NfieldInterviewsServiceTests.cs]: 0 as interviewId is not valid so do random from 1-9999

**Comment by neilboyd** [comment]
Write a short description of the issue rather than referring to the issue that's in our private repo.

File: Library/Models/Interview.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Nfield.Models
+{
+    /// <summary>
+    /// Holds all properties of an interviewer
```
Comment (neilboyd) [inline Library/Models/Interview.cs]: inaccurate summary

File: Library/Extensions/NfieldConnectionClientExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Recursive method that polls the activity status until it completes.
+        /// </summary>
+        /// <param name="client">NField Connection Client (this)</param>
+        /// <param name="activityId">The id of the activity to wait for.</param>
+        /// <param name="fieldNameResult">The name of the result field</param>
+        /// <returns></returns>
```
Comment (neilboyd) [inline Library/Extensions/NfieldConnectionClientExtensions.cs]: remove empty comment, or add a description

**Comment by carloscoceraperez** [comment]
@neilboyd  isn't useful to put related PRs references in the decriptions? Maybe this makes it easier to make reviewers look into other PRs to verify REPOs code interactions are fine.

**Comment by neilboyd** [comment]
This is a public repo and people can't access our private repo.

**Pull request merged**